libs/gl-testing/gltesting/scheduler.py

```diff
@@ -413,11 +419,15 @@ async def PairDevice(self, req: schedpb.PairDeviceRequest):
             "restrs": req.restrs
         }
         self.sessions[req.session_id] = data
+
+        # Wait for the Approval of an old device.
+        await self.recv_once(self.recv_stream)
```

Comment:
nit: Not checking the type here is likely an error: you are assuming a canonical ordering on the events that come in through this stream, but the way it is constructed suggests that the intent is to add more use-cases to theis stream incrementally as we need them. Adding new messages could invalidate this assumption, and concurrent pairing requests will definitely cause desyncs. Guarding by looping until we get the correct message is the safer way of doing this, or fall back to classical call-response patterns where this association is not needed.

This is minor for now, but a potential source for headaches further down the road, and having an assert or something here, with a brief note about the assumptions could help a lot.